aicsimageio/readers/reader.py

```diff
@@ -114,11 +115,36 @@ def is_this_type(cls, data: types.ImageLike) -> bool:
     def _is_this_type(buffer: io.BytesIO) -> bool:
         pass
 
-    @property
     @abstractmethod
-    def dask_data(self) -> da.core.Array:
+    def _build_delayed_dask_data(self) -> da.core.Array:
+        pass
+
+    @abstractmethod
+    def _read_in_memory_data(self) -> np.ndarray:
         pass
 
+    @property
+    def dask_data(self) -> da.core.Array:
+        if self._dask_data is None:
+            try:
+                # These lines both check if distributed has been imported
+                # and if a client connection has been created
+                # If distributed hasn't been imported it will KeyError
+                # If no client has been created it will ValueError
+                # We can't import distributed due to it requiring network utilities:
+                # https://github.com/AllenCellModeling/aicsimageio/issues/82
+                sys.modules["distributed"].get_client()
+
+                # No error means there is a cluster and client
+                # available on this worker process
+                # Use delayed dask reader
+                self._dask_data = self._build_delayed_dask_data()
+            except (KeyError, ValueError):
+                self._data = self._read_in_memory_data()
+                self._dask_data = da.from_array(self._data)
```

Comment:
controversial suggestion:  rename `_build_delayed_dask_data` to `_read_delayed` and rename `_read_in_memory_data` to `_read_immediate`

Reply:
I like those names.